pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -155,6 +155,12 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying ShutdownGracePeriod or ShutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
 	}
+	if localFeatureGate.Enabled(features.GracefulNodeShutdownBasedOnPodPriority) && len(kc.ShutdownGracePeriodByPodPriority) != 0 && (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) {
+		allErrors = append(allErrors, fmt.Errorf("invalid configuration: Cannot specify both ShutdownGracePeriodByPodPriority and ShutdownGracePeriod at the same time"))
+	}
+	if !localFeatureGate.Enabled(features.GracefulNodeShutdownBasedOnPodPriority) && len(kc.ShutdownGracePeriodByPodPriority) != 0 {
```

Comment:
```suggestion
	if localFeatureGate.Enabled(features.GracefulNodeShutdownBasedOnPodPriority) {
		if len(kc.ShutdownGracePeriodByPodPriority) != 0 && (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) {
			allErrors = append(allErrors, fmt.Errorf("invalid configuration: Cannot specify both ShutdownGracePeriodByPodPriority and ShutdownGracePeriod at the same time"))
		}
		if len(kc.ShutdownGracePeriodByPodPriority) != 0 {
```

I agree we don't want nested verifications and that we should avoid else clauses here, but if we lift out the feature gate verification it will make it easier when we remove the feature gate later.